terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -1821,7 +1821,7 @@ def qq(*words):
                     },
                     'statistic': stat,
                     'threshold': threshold,
-                    'treat_missing_data': 'missing',
+                    'treat_missing_data': 'notBreaching',
                     **{
                         state + '_actions': ['${data.aws_sns_topic.monitoring.arn}']
                         for state in ('insufficient_data', 'alarm', 'ok')
```

Comment:
If missing data isn't treated as missing then we shouldn't have an alarm for `insufficient_data`. However, I was under the impression that this was added for an alarm to go off when the instance isn't producing any data, hence the insufficient data alarm. Does the data points to alarm parameter apply differently for `insufficient_data` type of alarms?

Reply:
> If missing data isn't treated as missing then we shouldn't have an alarm for `insufficient_data`.

Good point, removed the `insufficient_data_alarm_actions`.

> However, I was under the impression that this was added for an alarm to go off when the instance isn't producing any data, hence the insufficient data alarm.

I can't find this functionality requested in the [original ticket](https://github.com/DataBiosphere/azul/issues/5139) or [PR](https://github.com/DataBiosphere/azul/pull/5275).

It seems that the `INSUFFICIENT_DATA` state is unavoidable when the alarm is first created if missing data is treated as `missing`, regardless of the `datapoints_to_alarm` and `evaluation_periods` values.

Likewise if missing data is treated as `breaching`, the alarm will enter an `ALARM` state upon creation. (Example: Alarm [email](https://groups.google.com/a/ucsc.edu/g/azul-group/c/nQ9ksXJ4iYM/m/HZolpDFgAwAJ) from the creation of the trail_logs alarm )

Since updating Gitlab causes these GitLab alarms to be recreated each time, I propose for these alarms we do not alarm on missing data.

I'll request this for a parking lot discussion.